src/main/java/com/group8/library_management/service/impl/UserServiceImpl.java

```diff
@@ -12,4 +22,53 @@ public UserServiceImpl(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
 
+    @Override
+    public Page<User> getAllUsers(int page, int size, String searchName, String sortBy) {
+        Sort sort = Sort.unsorted();
+        if (sortBy != null && !sortBy.isEmpty()) {
+            if (sortBy.contains(",")) {
+                String[] parts = sortBy.split(",");
+                sort = Sort.by(Sort.Direction.fromString(parts[1].toUpperCase()), parts[0]);
+            } else {
+                sort = Sort.by(sortBy).ascending();
+            }
+        } else {
+            // Sắp xếp mặc định nếu không có tùy chọn sortBy được cung cấp
+            sort = Sort.by("id").ascending();
+        }
```

Comment:
```suggestion
        // Xây dựng đối tượng Sort, ưu tiên sortBy nếu được cung cấp
    Sort sort = Optional.ofNullable(sortBy)
            .filter(s -> !s.isEmpty())
            .map(s -> {
                if (s.contains(",")) {
                    String[] parts = s.split(",");
                    return Sort.by(Sort.Direction.fromString(parts[1].trim().toUpperCase()), parts[0].trim());
                } else {
                    return Sort.by(Sort.Order.asc(s.trim()));
                }
            })
            .orElse(Sort.by(Sort.Order.asc("id")));
```

refactor theo 2 tiêu chí:
Sử dụng Optional để kiểm tra và xử lý giá trị sortBy giúp code ngắn gọn, tránh lỗi NullPointerException.
Sort.Order.asc(...) là cú pháp mới và rõ ràng của Spring Data 3.x.